### PR TITLE
[#5930] JSON schemas now conform to 2020-12 specification (main)

### PR DESCRIPTION
**Pull request by korydraughn**
The PR attempts to accomplish the following things:
- (**NOT TESTED**) Schema files are configured to use `IRODS_HOME_DIRECTORY` when building
  from source. This enables validation support for non-package installations.
- Schema validation now supports absolute and relative paths.
    - Verified that http(s) is supported as well.
- Removed v3 schema directory.

### CODE REVIEW

File: server/CMakeLists.txt [hunk 1]
```diff
@@ -109,6 +110,13 @@ foreach(SCHEMA_FILE IN LISTS IRODS_CONFIGURATION_SCHEMA_FILES)
     COMPONENT ${IRODS_PACKAGE_COMPONENT_SERVER_NAME}
   )
 endforeach()
+# TODO Is this way of updating the ids preferred over using CMake's configure_file() function?
+# If we use input files for the schema files, we can remove these lines and configuration_schemas/update_schema_ids_for_cmake.py.
+# Given that the schemas are maintained with the iRODS source, it would seem we don't need to do this.
+#
+# Q. Is the python script used outside of this location?
+# Q. Given that the configuration_schemas directory contains a README file, should we assume users run the python script directly?
+# Q. Are there any advantages to using this over configure_file()?
```
Comment (korydraughn) [inline server/CMakeLists.txt:119]: @trel @SwooshyCueb Any thoughts on this?

File: configuration_schemas/v4/VERSION.json.in [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "id": "file:///var/lib/irods/configuration_schemas/v4/VERSION.json",
+    "$id": "file:///@IRODS_HOME_DIRECTORY@/configuration_schemas/v4/VERSION.json",
```
Comment (korydraughn) [inline configuration_schemas/v4/VERSION.json.in:2]: @SwooshyCueb Is this enough for non-package installs? Should `CMAKE_INSTALL_PREFIX` be used here as well?

I noticed that it is used with `IRODS_HOME_DIRECTORY` in server/CMakeLists.txt (see https://github.com/irods/irods/pull/6275/files#r835943065) to update the schema files.

I know that `file:////` appeared when I tried using it and the validator didn't have any issues. I'm guessing that is because `////` is collapsed into `///`.

Comment (korydraughn) [reply configuration_schemas/v4/VERSION.json.in:2]: The `file:///` results in an absolute path rather than a relative path.

For people not familiar with this, `file:///` is equivalent to `file://localhost/`.

See https://en.wikipedia.org/wiki/File_URI_scheme for more details.

File: server/CMakeLists.txt (hunk 1 above)
Comment (korydraughn) [reply server/CMakeLists.txt:119]: Also, `update_schema_ids_for_cmake.py` is not referenced anywhere else in this repo.

File: configuration_schemas/v4/VERSION.json.in (hunk 2 above)
Comment (SwooshyCueb) [reply configuration_schemas/v4/VERSION.json.in:2]: Honestly I think having absolute paths written down anywhere is problematic. Not just for non-package installs, but also for relocatable packages and tarball packages.

File: server/CMakeLists.txt (hunk 1 above)
Comment (SwooshyCueb) [reply server/CMakeLists.txt:119]: Like I said in my other comment, I think ideally we shouldn't be doing this at all. Instead taking the time to switch over to `configure_file`, we should instead put that effort towards eliminating the need for absolute paths in our schema ids.

Comment (korydraughn) [reply server/CMakeLists.txt:119]: See other comment.

File: configuration_schemas/v4/VERSION.json.in (hunk 2 above)
Comment (korydraughn) [reply configuration_schemas/v4/VERSION.json.in:2]: I'll create an issue about absolute paths and we can go on and merge this.

Comment (korydraughn) [reply configuration_schemas/v4/VERSION.json.in:2]: Done.

**Comment by korydraughn** [comment]
Given the time constraints, we've agreed that attempting to address #6283 is out-of-scope for 4.3.0 and will be handled in a later release.

**Comment by korydraughn** [comment]
The final task for this PR is to verify that non-package installs work.

**Comment by korydraughn** [comment]
Thinking about this more, verifying correctness of non-package installs is not the focus of this PR.

We should handle that separately. For that reason, I've created a new issue: #6284

**Comment by korydraughn** [comment]
Will run it through the test suite and add the #'s

File: scripts/irods/json_validation.py [hunk 3]
```diff
@@ -113,10 +114,11 @@ def get_initial_schema_from_web(schema_uri):
...
     return schema
 
-def get_initial_schema_from_file(schema_uri):
-    with open(schema_uri[7:], 'rt') as f:
-        schema = json.load(f)
-    return schema
+def load_json_schema_from_file(schema_uri):
+    # Skipping the first six characters (i.e. the "file://" prefix) results
+    # in either a relative path or an absolute path. 
+    with open(schema_uri.strip()[6:], 'rt') as f:
```
Comment (trel) [inline scripts/irods/json_validation.py:120]: why the change from 7 to 6?    was it ... just wrong before?

`file://` is definitely seven characters...

Comment (korydraughn) [reply scripts/irods/json_validation.py:120]: It's zero indexed

Comment (trel) [reply scripts/irods/json_validation.py:120]: so... it was just wrong when set to 7?     sheesh.

Comment (korydraughn) [reply scripts/irods/json_validation.py:120]: This fixes support for relative and absolute paths

Comment (trel) [reply scripts/irods/json_validation.py:120]: Where it didn't before.   Okay.  Thanks.

**Comment by korydraughn** [comment]
All core tests passed.

Added #'s
